grist-doc-wk-exporter/main.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+import psutil
+from prometheus_client import start_http_server
+from prometheus_client.core import (
+    CollectorRegistry,
+    CounterMetricFamily,
+    GaugeMetricFamily,
+)
+from prometheus_client.registry import Collector
+
+METRICS_PORT = int(os.environ.get("METRICS_PORT", 9090))
+
+
+def get_tini_process():
+    for proc in psutil.process_iter():
+        try:
+            cmdline = proc.cmdline()
+        except Exception as e:
+            print(f"Could not retrieve cmdline: {e}")
+            continue
+        if "/usr/bin/tini" in cmdline:
+            return proc
+
+
+class CustomCollector(Collector):
+    def collect(self):
+        # To read correct cgroup we need to read the root fs of the grist init process
+        tini_process = get_tini_process()
```

Comment:
This could return `None`, do you want to catch this case ?

Reply:
I didn't have a better idea of how to handle this. If there's no tini process, then I'm not sure how we can find a filesystem root in the correct namespace. I mean we could fallback to finding the node process, but that could return none as well, or find a python process, which could return none as well... I chose to avoid that complexity. Ideally there would be a way to be sure to access the other container's root fs but I haven't found one yet